src/main/java/seedu/address/logic/parser/TimeParser.java

```diff
@@ -172,6 +170,131 @@ public static List<Interview> listInterviewsToday(UniqueInterviewList interviews
         return res;
     }
 
+    /**
+     * Compiles a list of free times that the user has. Each element is a 2-element list where the
+     * first element is the start of the free time block, and the second element is the end of the
+     * free time block. Only places interviews that are within a 9-5 workday. Assumes that the given
+     * interview list has no clashes. Also assumes that the start time of any scheduled interviews are less
+     * than or equals to their corresponding end time.
+     *
+     * @author Tan Kerway
+     * @param day the day that the user inputs
+     * @param interviewList the list of interviews that the user has
+     * @return a list of free time blocks that the user has on a given day
+     */
+    public static List<List<LocalDateTime>> listPocketsOfTimeOnGivenDay(
```

Comment:
this method is rather lengthy, perhaps considering breaking into a few function for greater readability